src/debuggability.js

```diff
@@ -161,6 +163,46 @@ Promise.prototype._propagateFrom = function (parent, flags) {
     USE(flags);
 };
 
+Promise.monitor = function () {
+    // Pending promises are stored in array for better performance: number
+    // of simultaneously pending promises is assumed to be relatively small,
+    // thus storing promises in the map and calculating hash for each promise
+    // is more expensive then iterating trough pending promises array when
+    // each promise is finished
+    Promise.prototype._pendingPromises = [];
+
+    function registerPromise() {
+        Promise.prototype._pendingPromises.push(this);
+    }
+
+    function unregisterPromise() {
+        var indexOfPromise = Promise.prototype._pendingPromises.indexOf(this);
+        ASSERT(indexOfPromise >= 0);
+        Promise.prototype._pendingPromises.splice(indexOfPromise,1);
+    }
+
+    util.wrapMethodIfExistsDefineIfNot(Promise.prototype,"_promiseCreated",
+        registerPromise);
+    util.wrapMethodIfExistsDefineIfNot(Promise.prototype,"_promiseSettled",
+        unregisterPromise);
+
+    Promise.getAllPendingPromises = function () {
+        return Promise.prototype._pendingPromises;
```

Comment:
I'd like to see a `.slice()` here, returning arrays that are later mutated is error-prone.
Or maybe don't make this a `get…` method at all, but just a public `Promise.pendingPromises` property (which can be accessed and where it's clear that the array is mutable)